aloha-core/src/main/scala/com/eharmony/aloha/models/exploration/BootstrapModel.scala

```diff
@@ -0,0 +1,138 @@
+package com.eharmony.aloha.models.exploration
+
+import com.eharmony.aloha.factory.{ModelFactory, ModelParser, ParserProviderCompanion}
+import com.eharmony.aloha.id.ModelIdentity
+import com.eharmony.aloha.models.{BaseModel, Model}
+import com.eharmony.aloha.score.Scores.Score
+import com.eharmony.aloha.score.basic.ModelOutput
+import com.eharmony.aloha.score.conversions.ScoreConverter
+import com.eharmony.aloha.score.conversions.ScoreConverter.Implicits.IntScoreConverter
+import com.eharmony.aloha.semantics.Semantics
+import com.eharmony.aloha.semantics.func.GenAggFunc
+import com.mwt.explorers.BootstrapExplorer
+import com.mwt.policies.Policy
+
+import scala.annotation.tailrec
+import scala.collection.JavaConversions.seqAsJavaList
+import scala.collection.{immutable => sci}
+
+/**
+  * Since explore-java has chosen to force the Policy to evaluate we can't just evaluate the model inside of the policy.
+  * We have to invert the control and evaluate the models first, and only upon all models successfully evaluating
+  * can we then do the policy evaluation.  Because of this IOC the policy becomes a simple lookup into the model map.
+  * @param index the model to get the action for
+  */
+private[this] case class NumberedPolicy(index: Int) extends Policy[sci.IndexedSeq[Int]] {
+  override def chooseAction(actions: sci.IndexedSeq[Int]): Int = actions(index)
+}
+
+/**
+  * A model for performing bootstrap style exploration.  This makes use of a number of policies.  The algorithm chooses
+  * one policy and then uses the other to calculate the appropriate probability of choosing that action.
+  * @param modelId a model identifier
+  * @param models a set of models that generate Int's.  These models MUST be deterministic for the probability to be correct.
```

Comment:
There should be a note in the comment for `models` that each model should return a value between 1 and `classLabels.size` inclusive.  I see it in `classLabels` but it is definitely better here.  The same goes for the epsilon-greeedy model.